File: frontend/src/microservices/tour-group/objects/CreateTourGroupPlannerRequest.ts
// This file defines `tour-group` module `CreateTourGroupPlannerRequest` as request payload data and provides JSON helpers.

export type CreateTourGroupPlannerRequest = {
  organizerUserId: string
  title: string
  description: string
  destination: string
  startDate: string
  endDate: string
  capacity: number
  coverImageUrl: string | null
  tags: string[]
}
export const createTourGroupPlannerRequestFromJson = (json: string): CreateTourGroupPlannerRequest =>
  JSON.parse(json) as CreateTourGroupPlannerRequest

export const createTourGroupPlannerRequestToJson = (value: CreateTourGroupPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/CreateTourGroupPlannerRequest.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateTourGroupPlannerRequest(
    organizerUserId: String,
    title: String,
    description: String,
    destination: String,
    startDate: String,
    endDate: String,
    capacity: Int,
    coverImageUrl: Option[String] = None,
    tags: List[String] = Nil
)
object CreateTourGroupPlannerRequest:
  given sourceEncoder: Encoder[CreateTourGroupPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateTourGroupPlannerRequest] = deriveDecoder
